web/source/text/prediction/modelManager.ts

```diff
@@ -49,33 +95,73 @@ namespace com.keyman.text.prediction {
 
     private unloadModel() {
       this.lmEngine.unloadModel();
-      this.currentModel = null;
+      delete this.currentModel;
+      delete this.configuration;
     }
 
-    private loadModel(model: ModelSpec) {
+    private loadModel(model: ModelSpec): Promise<void> {
       if(!model) {
         throw new Error("Null reference not allowed.");
       }
 
       let file = model.path;
-      this.lmEngine.loadModel(file);
-      this.currentModel = model;
+      let mm = this;
+
+      // We should wait until the model is successfully loaded before setting our state values.
+      return this.lmEngine.loadModel(file).then(function(config: Configuration) { 
```

Comment:
same as `return mm.lmEngine.loadModel(file)...` ?

Reply:
`loadModel(file)` returns a `Promise` that will 'return' asynchronously.  The `then` block is automatically handled upon its completion.